src/test/java/orlov/programming/spring_core_gym/service/user/TraineeServiceTest.java

```diff
@@ -0,0 +1,193 @@
+package orlov.programming.spring_core_gym.service.user;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import orlov.programming.spring_core_gym.dao.impl.user.DAOUsernameFindable;
+import orlov.programming.spring_core_gym.model.user.Trainee;
+import orlov.programming.spring_core_gym.util.PasswordGenerator;
+
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class TraineeServiceTest {
```

Comment:
for test with mockito you can `verify` to check that mock was invoked
https://www.baeldung.com/mockito-verify